resource_detectors/test/opentelemetry/detectors/deployment_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+describe OpenTelemetry::Resource::Detectors::Deployment do
+  let(:detector) { OpenTelemetry::Resource::Detectors::Deployment }
+
+  describe '.detect' do
+    let(:detected_resource) { detector.detect }
+    let(:detected_resource_attributes) { detected_resource.attribute_enumerator.to_h }
+    let(:expected_resource_attributes) { {} }
+
+    it 'returns an empty resource' do
+      _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
+      _(detected_resource_attributes).must_equal(expected_resource_attributes)
+    end
+
+    describe 'when in a rack environment' do
+      it 'returns a resource with rack environment' do
+        old_env = ENV['RACK_ENV']
+        ENV['RACK_ENV'] = 'env from test'
+        begin
+          _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
+          _(detected_resource_attributes).must_equal('deployment.environment' => 'env from test')
+        ensure
+          ENV['RACK_ENV'] = old_env
+        end
+      end
```

Comment:
Please consider adding a test for the `Rails.env` case. 

You can use helpers like `stub_const` to simulate the existence of classes:

https://github.com/open-telemetry/opentelemetry-ruby/search?q=stub_const

Reply:
Added it like this:
```rb
      it 'return rails env when present' do
        allow(::Rails).to receive(:env).and_return('env from rails')
        with_env({}) do
          _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
          _(detected_resource_attributes).must_equal('deployment.environment' => 'env from rails')          
        end
      end
```

Reply:
Looks like this test is failing in CI.

You'll have to stub the `Rails` constant since it doesn't exist. 

```
NameError: uninitialized constant Rails
5431
    /home/runner/work/opentelemetry-ruby/opentelemetry-ruby/resource_detectors/test/opentelemetry/detectors/deployment_test.rb:22:in `block (4 levels) in <top (required)>'
5432
```

Reply:
done. tests should pass now :)